application/models/QuestionAttribute.php

```diff
@@ -578,4 +589,39 @@ public static function getOwnQuestionAttributesViaPlugin()
 
         return (array) $result->get('questionAttributes');
     }
+
+    /**
+     * Apply XSS filter to question attribute value unless 'xssfilter' property is false.
+     * @param string $attribute the name of the attribute to be validated.
+	 * @param array<mixed> $params additional parameters passed with rule when being executed.
+     * @return void
+     */
+    public function filterXss($attribute, $params)
+    {
+        $question = Question::model()->find("qid=:qid", ['qid' => $this->qid]);
+        if (empty($question)) {
+            return;
+        }
+        $questionAttributeFetcher = new \LimeSurvey\Models\Services\QuestionAttributeFetcher();
+        $questionAttributeFetcher->setQuestion($question);
+        $questionAttributeDefinitions = $questionAttributeFetcher->fetch();
+
+        // The value will be filtered unless the attribute definition has the "xssfilter" property set to false
+        $shouldFilter = true;
+        if (isset($questionAttributeDefinitions[$this->attribute])) {
+            $questionAttributeDefinition = $questionAttributeDefinitions[$this->attribute];
```

Comment:
No way to remove the `array_key_exists` check here ? Adding xssFilter => true on https://github.com/LimeSurvey/LimeSurvey/blob/b43948123d3cb6e1a029779df0b346a704af3a5a/application/models/QuestionAttribute.php#L355 didn't work ?

(If not it's really another dev issue)

Reply:
OK, we will add the xssFilter => TRUE on `getDefaultSettings() `.
Still, I think we should still check, just in case, to avoid problems with old plugins, question themes ... or not expected situations...

Reply:
>  to avoid problems with old plugins, question themes

Reason to have default settings, it' added for this : be sure to have it set.

> or not expected situations...

a not expected situation means a potential issue : i prefer to have an error at debug > 0 in this case.

Reply:
The problem is that with debug = 0 the error will not show and there is the possibility that the attribute doesn't get  filtered when supposed to. We added the xssFilter on defaultSettings, but prefer to leave the `array_key_exists`, if it is ok. Thanks!